apps/o2-blocks/src/github-issue-template-button/editor.js

```diff
@@ -0,0 +1,158 @@
+
+registerBlockType( 'a8c/github-issue-template-button', {
+	title: __( 'Github Isssue Template Button', 'a8c' ),
+	icon,
+	category: 'layout',
+	attributes: {
+		userOrOrg: {
+			type: 'string',
+			default: 'automattic',
+		},
+		repo: {
+			type: 'string',
+		},
+		title: {
+			type: 'string',
+		},
+		body: {
+			type: 'string',
+		},
+		buttonLabel: {
+			type: 'string',
+			default: 'Create Issue',
+		},
+		alignment: {
+			type: 'string',
+		},
+	},
+	edit: ( props ) => {
+		const {
+			attributes: { userOrOrg, repo, title, body, buttonLabel, alignment },
+		} = props;
+
+		const alignmentOptions = [
+			{ key: 'flex-start', name: 'Left' },
+			{ key: 'center', name: 'Center' },
+			{ key: 'flex-end', name: 'End' },
+		];
+
+		const onChangeUserOrOrg = ( newUserOrOrg ) => {
+			props.setAttributes( { userOrOrg: newUserOrOrg } );
+		};
+
+		const onChangeRepoName = ( newRepo ) => {
+			props.setAttributes( { repo: newRepo } );
+		};
+
+		const onChangeTitle = ( newTitle ) => {
+			props.setAttributes( { title: newTitle } );
+		};
+
+		const onChangeBody = ( newBody ) => {
+			props.setAttributes( { body: newBody } );
+		};
+
+		const onChangeButtonLabel = ( newButtonLabel ) => {
+			props.setAttributes( { buttonLabel: newButtonLabel } );
+		};
+
+		const onChangeAlignment = ( control ) => {
+			props.setAttributes( { alignment: control.selectedItem.key } );
+		};
+
+		return (
+			<Fragment>
+				<div className={ `wp-block-github-issue-template-button ${ alignment }` }>
+					<Button isPrimary>
+						<Icon icon={ icon } />
+						&nbsp;{ buttonLabel }
+					</Button>
+				</div>
+				<InspectorControls>
+					<TextControl
+						label={ __( 'User or Organization' ) }
+						onChange={ onChangeUserOrOrg }
+						value={ userOrOrg }
+					/>
+					<TextControl
+						label={ __( 'Repository Name' ) }
+						onChange={ onChangeRepoName }
+						value={ repo }
+					/>
+					<TextControl label={ __( 'Issue Title' ) } onChange={ onChangeTitle } value={ title } />
+					<TextareaControl label={ __( 'Issue body' ) } onChange={ onChangeBody } value={ body } />
+					<TextControl
+						label={ __( 'Button Label' ) }
+						onChange={ onChangeButtonLabel }
+						value={ buttonLabel }
+					/>
+					<CustomSelectControl
+						label={ __( 'Alignment' ) }
+						options={ alignmentOptions }
+						value={
+							alignmentOptions.find( ( option ) => option.key === alignment ) ||
+							alignmentOptions[ 0 ]
+						}
+						onChange={ onChangeAlignment }
+					/>
+				</InspectorControls>
+			</Fragment>
+		);
+	},
+	save: ( props ) => {
+		let url;
+
+		const {
+			attributes: { userOrOrg, repo, title, body, buttonLabel, alignment },
+		} = props;
+
+		if ( repo ) {
+			url = newGithubIssueUrl( {
+				title,
+				repo,
+				user: userOrOrg,
+				body,
+			} );
+		}
+
+		return (
+			<div className="block" style={ `display: flex; justify-content: ${ alignment }` }>
+				<Icon icon={ icon } />
+				&nbsp;
+				<Button isPrimary href={ url || '' }>
```

Comment:
🤔  This is effectively a link to the page we're already on, isn't it? I wonder if it makes sense to render anything at all if we can't make a good link. Maybe we shouldn't render anything (`null`) or show a message. What do you think?

Reply:
Yeah, that is the link that is saved with the block. Thanks for pointing that out, this was a scenario that I did want to discuss with you, the short-circuit was added as a temporary thing as I didn't want `undefined` to sneak in. 

We could explore the following solutions:

1. Add validation in the settings panel to make sure that all the required fields for the URL are filled by the user. I'm not sure how we would implement the actual validation in the context of the block editor or if its block API supports it though, so I'd need to do some further research. Can we prevent the page/block from being saved if one of its blocks is missing critical data?
1. If `1` is not possible, then the at the time the page/block is saved, your suggestion of showing a message in the frontend telling the user the button needs to be properly setup makes sense. I just don't know how to present it, yet. Maybe an alert upon clicking it?
1. Should we also add any visual cues to the button in the editor view if it's not properly setup? Maybe some kind of warning icon with a tooltip somewhere around the block area?

Reply:
> Add validation in the settings panel to make sure that all the required fields for the URL are filled by the user. I'm not sure how we would implement the actual validation in the context of the block editor or if its block API supports it though, so I'd need to do some further research.

As long as we have a repo, we can render a link, right? Shall we add an error message in the case that the repo hasn't been provided?

> Can we prevent the page/block from being saved if one of its blocks is missing critical data?

The block will be saved, but we can prevent it from rendering anything by adding an early return to save:

```ts
if ( ! repo ) {
  return null;
}
```

Reply:
> As long as we have a repo, we can render a link, right? Shall we add an error message in the case that the repo hasn't been provided?

I think it does make sense to render an error message if a link is not provided. I assume you mean using the actual block rendering area in the editor for that. I'll try to come up with a nice way to present that error message there. Perhaps there's already an established UI pattern for showing block error messages, do you know?

If not, I don't think we need to complicate much. Maybe showing a warning icon + a text in place of the button if the right setting is not set, something like:

:warning: This button needs to be configured!
_Please, configure it by selecting it and adding at least a repo where the issue will be created_

Once the required data is inputted, then it renders the button as expected, instead of the message. In the frontend, the button won't be shown if the required fields were not provided (by preventing the markup from being saved, as you suggested above).